src/main/java/htsjdk/variant/vcf/VCFCompoundHeaderLine.java

```diff
@@ -160,9 +163,11 @@ protected VCFCompoundHeaderLine(String line, VCFHeaderVersion version, Supported
         super(lineType.toString(), "");
 
         final ArrayList<String> expectedTags = new ArrayList(Arrays.asList("ID", "Number", "Type", "Description"));
-        if (version.isAtLeastAsRecentAs(VCFHeaderVersion.VCF4_2))
-            expectedTags.add("Version");
-        final Map<String, String> mapping = VCFHeaderLineTranslator.parseLine(version, line, expectedTags);
+        Set<String> optionalTags = Collections.emptySet();  
+        if (version.isAtLeastAsRecentAs(VCFHeaderVersion.VCF4_2)) {
+            optionalTags = new HashSet<String>(Arrays.asList("Source", "Version"));
```

Comment:
```suggestion
            optionalTags = new HashSet<>(Arrays.asList("Source", "Version"));
```

Reply:
Also, not sure of the benefit vs overhead of using `Set` vs `List` here if the collection only contains two elements.

Reply:
I was trying to make an interface that can survive into the future as the number of flag grows.
The optional flags are being tested for presence/absence, their order does not matter in this case, which is a Set operation. If you think this is extremely unlikely to grow to the point when a Set would matter, I can switch back to list. Would not want to have a breaking change in the interface later.